Initial inet_ntop implementation

dalehamel opened the pull request:
This is ready for review.

I have implemented `ntop` as described in #30, but with a few limitations:

* Only ipv4 addresses are supported. The main reason ipv6 addresses aren't supported is that there is no way to pass them in right now, is they are stored as a 4x32 bit array in kernel, and there is no way to assign this type to a variable in bpftrace in order to pass it in
* The constants `AF_INET` and `AF_INET6` would need to be added as builtin idents to bpftrace, in order to specify them as the first argument, as the `#define` macros from the kernel headers won't be pulled in as any sort of usable variable (and probably shouldn't). We could define them builtin idents pretty easily using a similar approach to `nsecs`, but for now you must specify the literal 2 for AF_INET, and the literal 10 for (currently unsupported) AF_INET6

The current way to use this function is as so:

```
./bpftrace -e 'BEGIN{ printf("%s\n", ntop(2, 0xFFFFFFFF)); }'
Attaching 1 probe...
255.255.255.255
```

* [x] ~Add tests for printing ipv4 addresses~ this sort of functional test would require a new test harness, not sure if worth the effort. I added what tests I could
* [x] Add map support to be able to use ipv4 addresses in map logic
* [x] ~Codegen tests should probably be added, but test suit is failing... #187 ~ Added for use as map key, but nothing else.
* [x] Write llvm IR to copy bytes to userspace buffer properly

dalehamel commented:
Basic working implementation:

```
 ./bpftrace -e 'BEGIN { printf("%s\n", inet_ntop(0x6c39950a)); }'
Attaching 1 probe...
```

dalehamel commented:
@ajor when you have a minute, review appreciated :)

brendangregg commented:
I had a look, looks good. Only two minor comments:

- this error: "Only AF_INET ipv4 address family is supported by ntop." -- should say "currently", eg, "ntop() currently only supports AF_INET (IPv4); IPv6 will be supported in the future."
- ntop() should be included in the reference guide.

Plus collapse the commits. Thanks!

dalehamel commented:
I've collapsed the commits, updated the reference guide with usage and a couple of examples, and added the note to the warning message if trying to parse an ipv6 address.

brendangregg commented:
Thanks, looks good to me, I'll merge.

The tests currently have 45 failed tests, but I don't believe this PR caused them, and I'm hoping #292 cleans up these failed tests.
